lib/Cake/Test/Case/Model/CakeSchemaTest.php

```diff
@@ -650,6 +683,34 @@ public function testSchemaReadWithCrossDatabase() {
 	}
 
 /**
+ * testSchemaRead method when a primary key is on a non-conventional column
+ *
+ * @return void
+ */
+	public function testSchemaReadWithNonConventionalPrimaryKey() {
+		$this->skipIf($this->db instanceof Postgres, 'Cannot test on Postgres');
+		$db = ConnectionManager::getDataSource('test');
+		$fixture = new NonConventionalPrimaryKeyFixture();
+		$fixture->create($db);
+
+		$read = $this->Schema->read(array(
+			'connection' => 'test',
+			'name' => 'TestApp',
+			'models' => false
+		));
+		$fixture->drop($db);
+
+		$hasTable = isset($read['tables']['non_conventional']);
+		$this->assertTrue($hasTable, 'non_conventional table should appear');
+		$versionIdHasKey = $hasTable && isset($read['tables']['non_conventional']['version_id']['key']);
```

Comment:
You don't need this && here. If hasTable is not true the test will fail and no further assertions will be run.